irohad/model/execution/impl/domain_name_parser.cpp

```diff
@@ -0,0 +1,32 @@
+#include "model/execution/domain_name_parser.hpp"
+
+#include <boost/spirit/include/qi.hpp>
+#include <boost/spirit/include/qi_char_class.hpp>
+#include <boost/spirit/include/qi_lexeme.hpp>
+#include <boost/spirit/include/qi_optional.hpp>
+#include <boost/spirit/include/qi_parse.hpp>
+#include <boost/spirit/include/qi_plus.hpp>
+
+namespace iroha {
+  namespace model {
+    bool DomainNameParser::isValid(const std::string &str) {
+      namespace qi = boost::spirit::qi;
+
+      qi::rule<decltype(str.begin())> letter, let_dig, end_with_let_dig,
+          let_dig_hyp, label, subdomain, domain;
+      domain = subdomain | ' ';
```

Comment:
Does this allow `" "` as domain name? From rfc is fine, but maybe not a good thing here.
Any thoughts on that?

Reply:
Maybe we should follow the RFC 1123 and disallow a whitespace.
https://serverfault.com/questions/342284/does-active-directory-support-dns-names-with-spaces

Reply:
We should disallow whitespaces. Also, it might be nice to support katakana.

Reply:
Why do we need to support katakana?
Could you illustrate the use case of katakana?

Reply:
Using domain names written in katakana, like アルトプラス. We have a good Japanese community and it is not such a big alphabet, so it might be interesting to support it. The project is called Iroha after all ^^